### PR TITLE
[PORT] Gives announcements a makeover!

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

Ports the following PRs from /tg/:
- https://github.com/tgstation/tgstation/pull/79071
- https://github.com/tgstation/tgstation/pull/79315
- https://github.com/tgstation/tgstation/pull/79711
- https://github.com/tgstation/tgstation/pull/80403
- https://github.com/tgstation/tgstation/pull/80741
- https://github.com/tgstation/tgstation/pull/80849
- https://github.com/tgstation/tgstation/pull/81056
- https://github.com/tgstation/tgstation/pull/85083
and prolly some more i can't be bothered to dig up bc I just copy-pasted from latest /tg/ master and fixed it up

## Why It's Good For The Game

mmm eyecandy, makes them easier to notice and see / harder to miss

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

2024-10-08 (1728423039) ~ dreamseeker

</details>

## Changelog
:cl: Absolucy, LT3, san7890, Melbert, Aylong
add: Gave announcements an eyecandy makeover!
/:cl:

### CODE REVIEW

File: code/__HELPERS/priority_announce.dm [hunk 1]
```diff
@@ -86,34 +116,104 @@
  * message - the message contents of the announcement.
  * title - the title of the announcement, which is often "who sent it".
  * alert - whether this announcement is an alert, or just a notice. Only changes the sound that is played by default.
- * from - who sent the announcement, in the case of communications consoles or the like.
  * html_encode - if TRUE, we will html encode our title and message before sending it, to prevent player input abuse.
- * players - optional, a list mobs to send the announcement to. If unset, sends to all players.
+ * players - optional, a list mobs to send the announcement to. If unset, sends to all palyers.
```
Comment (Tsar-Salat) [inline code/__HELPERS/priority_announce.dm]: Hope you aren't copy-pasting without looking.

File: code/__HELPERS/priority_announce.dm [hunk 2]
```diff
@@ -86,34 +116,104 @@
...
  * sound_override - optional, use the passed sound file instead of the default notice sounds.
+ * should_play_sound - Whether the notice sound should be played or not.
+ * color_override - optional, use the passed color instead of the default notice color.
  */
-/proc/minor_announce(message, title = "Attention:", alert, from, html_encode = TRUE, list/players, sound_override)
```
Comment (Tsar-Salat) [inline code/__HELPERS/priority_announce.dm:94]: Make sure to retain bee's unique captain messages containing the senders name. I see the from arg is deleted so I will make a comment here.

I tried porting this a few months back and that's a change I had to look out for.

Edit: found the PR
- https://github.com/BeeStation/BeeStation-Hornet/pull/2726

**Review by Tsar-Salat — CHANGES_REQUESTED** [review]

File: code/__HELPERS/priority_announce.dm (hunk 1 above)
Comment (Absolucy) [reply code/__HELPERS/priority_announce.dm]: me when i open PRs while sleep deprived

File: code/__HELPERS/priority_announce.dm (hunk 2 above)
Comment (Absolucy) [reply code/__HELPERS/priority_announce.dm:94]: am i crazy or is auth_id never fucking passed anywhere anyways

**Comment by Rukofamicom** [comment]
Don't forget to credit original authors in the CL!

**Comment by PowerfulBacon** [comment]
Oh these are beautiful, I've been wanting to get these across for ages

File: code/__HELPERS/priority_announce.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/__HELPERS/priority_announce.dm:94]: Yea you're right, looks like it got removed from everywhere that needed it at some point

Comment (PowerfulBacon) [reply code/__HELPERS/priority_announce.dm:94]: @Absolucy @Tsar-Salat Bumping for a decision to be made on this change. How important is it, how much do we care and is it something we can just accept as being removed for now considering it seems to have been accidentally removed at some point anyway.

File: code/__HELPERS/priority_announce.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/__HELPERS/priority_announce.dm]: you never actually pushed the commit that fixes this @Absolucy

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**